app/controllers/selections_controller.rb

```diff
@@ -1,18 +1,37 @@
+include SelectionsHelper
+
 class SelectionsController < ApplicationController
 
-  def add
-    @user = User.find(session[:user_id])
-    @station = Station.all
-    # figure this out
-    @selection = Selection.new(params:user_id, station_id)
+  def index
+    @selections = current_user.stations    #provides an array of station objects
+    @forecasts = []
+    get_forecast(@selections)
+    @unselected_stations = Station.all - @selections # updates list of available stations for user
   end
 
+  def show
+  end
 
-  def destroy
-
+  def new
+    @selection = Selection.new
   end
 
+  def create
+     station = Station.find(params[:id])
```

Comment:
I don't think there's any reason to load the Station record here unless you want to verify that it exits. In that case it would be a little more efficient with
```Station.exists?(params[:id])```

Reply:
Need both station_id and user_id to make a new Selection record.

Reply:
Yeah but you don't need to instantiate the Station. You already have the station id in `params[:id]`.

Reply:
If a comment out that line, the create method fails.

Reply:
`Selection.find_or_create_by(user_id: current_user.id, station_id: params[:id])` does not work?

Reply:
Ahh... indeed it does work.  And actually that's the way I have in the #destroy